pkg/agent/config/traffic_encap_mode_test.go

```diff
@@ -0,0 +1,260 @@
+package config
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetTrafficEncapModeFromStr(t *testing.T) {
+	tests := []struct {
```

Comment:
I suggest keeping the table declarations more concise, since these tests are all very simple:
```
tests := []struct{
		name    string
		mode    string
		expBool bool
		expMode TrafficEncapModeType
	}{
		{"encap-mode-valid", "enCap", true, 0},
		{"no-encap-mode-valid", "Noencap", true, 1},
		...
         }
```